fronts-client/src/components/feed/RecipeSearchContainer.tsx

```diff
@@ -56,11 +67,21 @@ export const RecipeSearchContainer = ({ rightHandContainer }: Props) => {
     chefSelectors.selectLastFetchOrder(state)
   );
 
+  const [page, setPage] = useState(1);
+
   const debouncedRunSearch = debounce(() => runSearch(), 750);
 
   useEffect(() => {
-    debouncedRunSearch();
-  }, [selectedOption, searchText]);
+    if (page > 1 && searchText.length === 0) runSearch(page);
+    else debouncedRunSearch();
+  }, [selectedOption, searchText, page]);
+
+  //const chefsPagination = IPagination | null;
+  const chefsPagination: IPagination | null = useSelector((state: State) =>
+    chefSelectors.selectPagination(state)
+  );
+
+  const hasPages = !!(chefsPagination && chefsPagination.totalPages > 1);
```

Comment:
I find this condition a bit hard to understand (possibly because I’m less used to javascript/typescript conventions!). Would the following be an equivalent condition, and if so could we use that instead?

```suggestion
  const hasPages = (chefsPagination != null) && (chefsPagination.totalPages > 1);
```

Reply:
Thanks @emdash-ie for suggesting another way of writing above conditional code, I think I might leave it as-is to maintain consistency of using `!!` negation that are there on other places in code too. Also it return strict boolean which is common way of doing in JS it seems . 
I would like to have view from @jonathonherbert to check on this.

Reply:
Sure, happy to stick with the approach taken elsewhere! I think the condition I’ve proposed would also strictly return a boolean as well though, since each of the comparisons should return a boolean, feeding into the `&&` to return a boolean. (JavaScript can be odd about types, but I don’t think it’ll turn a boolean into something else here, and I don’t think the `> 1` or `!= null` comparisons can return a non-Boolean value.)

Reply:
O yes absolutely, your code will be returning boolean too (thats why I mentioned first that thanks for suggesting another way of conditional code), i think my main emphasising was on "common" way of using `!!` in JS for returning boolean which might ask to keep the condition as-is. 
But will check with @jonathonherbert too.

Reply:
Yes, that seems a bit clearer to me than the double negation.

Reply:
Thanks @emdash-ie for revising, i will commit above suggestion then 👍  Thanks @jonathonherbert